src/mercury_engine_data_structures/formats/bfstm.py

```diff
@@ -0,0 +1,199 @@
+    IMA_ADPCM = 3
+
+StreamInfo = Struct(
+    "sound_encoding" / EnumAdapter(SoundEncodingEnum, Byte),
+    "loop" / Flag,
+    "num_channels" / Byte,
+    "num_regions" / Byte,
+    "sample_rate" / Int32ul,
+    "loop_start" / Int32ul,
+    "num_frames" / Int32ul,
+    "num_blocks" / Int32ul,
+    "block_byte_size" / Int32ul,
+    "block_sample_size" / Int32ul,
+    "last_unpadded_block_bytes" / Int32ul, # size of last block without padding in bytes
+    "last_unpadded_block_samples" / Int32ul, # size of last block without padding in samples
+    "last_padded_block_bytes" / Int32ul, # sizeof last block with padding in bytes
+    "seek_info_size" / Const(4, Int32ul),
+    "seek_sample_interval" / Int32ul,
+    "sample_data_flag" / Const(0x1F00, Int32ul),
+    "sample_data_offset" / RebuildableOffset(), # relative to 0x08 in DATA
+    "region_info_size" / RebuildableOffset(),
+    "region_info_flag" / Const(0, Int32ul),
+    "region_info_offset" / RebuildableOffset(),
+    "orig_loop_start" / Int32ul,
+    "orig_loop_end" / Int32ul, # same as num_frames?
+    "unk" / Int32ul, # float maybe? idk what this is, doesn't make sense as a float or int
+)
+
+InfoSection = Struct(
+    RebuildOffset(construct.this._._.header.info_offset),
+
+    "_start" / Tell,
+    "_magic" / Const(b"INFO"),
+    "_size_addr" / Tell,
+    "size" / RebuildableOffset(), # size
+    "_info_flag" / Const(0x4100, Int32ul), # int16 wiht 2b padding
+    "info_offset" / RebuildableOffset(),
+    "track_info_flag" / Int32ul, # int16 with 2b padding
+    "track_info_offset" / RebuildableOffset(),
+    "channel_info_flag" / Const(0x101, Int32ul),
+    "channel_info_offset" / RebuildableOffset(),
+    RebuildOffset(construct.this._.info_offset, construct.this._._start + 8),
+    "stream_info" / StreamInfo,
+    # TODO track info, unsure if any dread files use it
+    RebuildOffset(construct.this._.channel_info_offset, construct.this._._start + 8),
+    "channel_infos" / ReferenceTable(0x4102, ChannelInfo, 0x0300, DspAdpcmInfo),
+    AlignTo(0x20),
+
+    RebuildOffset(construct.this._._.header.size, construct.this._._start),
+    RebuildOffset(construct.this._._.header.info_size, construct.this._._start),
+    RebuildOffset(construct.this._.size, construct.this._._start),
+)
+
+Header = Struct(
+    "_magic" / Const(b"FSTM"),
+    Const(b"\xff\xfe"), # Byte-Order Mark, always uses little endian
+    "_header_size" / Const(0x40, Int16ul),
+    "version" / VersionAdapter("1024.6.0"),
+    "size" / RebuildableOffset(),
+    "_num_datablocks" / Const(3, Int32ul),
+    "_info_flag" / Const(0x4000, Int32ul),
+    "info_offset" / RebuildableOffset(),
+    "info_size" / RebuildableOffset(),
+    "_seek_flag" / Const(0x4001, Int32ul),
+    "seek_offset" / RebuildableOffset(),
+    "seek_size" / RebuildableOffset(),
+    "_data_flag" / Const(0x4002, Int32ul),
+    "data_offset" / RebuildableOffset(),
+    "data_size" / RebuildableOffset(),
+    "hdr_end" / Tell,
+    AlignTo(0x20),
```

Comment:
Unfortunately seems like there's a lot of alignment in this struct which makes it unlikely to be compilable. this specific example could be since header is a constant size though.